api/v1beta1/machinehealthcheck_webhook.go

```diff
@@ -165,8 +175,12 @@ func (m *MachineHealthCheck) validate(old *MachineHealthCheck) error {
 		)
 	}
 
-	if len(allErrs) == 0 {
-		return nil
+	if len(m.Spec.UnhealthyConditions) == 0 {
```

Comment:
This validation could be left up to kubebuilder, which is how we're doing it for MHCSpec today.